plugins/org.eclipse.xtend.ide.common/src/org/eclipse/xtend/ide/common/highlighting/XtendHighlightingCalculator.java

```diff
@@ -384,6 +384,7 @@ public void acceptSemanticLineBreak(int charCount, RichStringLiteral origin, boo
 		@Override
 		public void acceptTemplateLineBreak(int charCount, RichStringLiteral origin) {
 			resetCurrentOffset(origin);
+			acceptor.addPosition(currentOffset, charCount, TEMPLATE_LINE_BREAK);
```

Comment:
I can only speak about the semantics of semantic_lb and template_lb as it is on the interface of the IRichStringPartAcceptor. Here, a semantic linebreak may occur in the output depending on the content in the line. A template_lb is never written to the output.
In the highlighter, the term template_lb appears to describe a linebreak that will definitely be written to the output. A potentail_lb is semantic but may be skipped if the line is empty.